### ISSUE TITLE
How XGBoost sample_weights alternate the objective function formula?

### ISSUE THREAD

**Issue by geogreko** [dmlc/xgboost#10297]
From my understanding and reading  post https://github.com/dmlc/xgboost/issues/144

when sample_weights are used, then first and second gradient

g_i and h_i are multiplied with sample_weights so that for each instance i:

grad_xgb[i] = grad[i] * case_weight[i]
hess_xgb[i] = hess[i] * case_weight[i]

The 2016 paper ‘XGBoost: A Scalable Tree Boosting System’ states that,  
if we remove the constant terms, the objective function at step t is given in eq.3

![image](https://github.com/dmlc/xgboost/assets/76202645/4e52cedd-2b5b-4bb3-b4cf-0ebc84b88b70)

 
The question is: How eq.3 is alternated when using sample weights? 
Is it as simple as that:

(g_i*case_weight[i])*ft(xi)+1/2(h_i * case_weight[i]) (ft)power of 2(xi) + Ω(ft)

In other words, just substituting g_i, and h_i with grad[i] * case_weight[i] and hess[i] * case_weight[i]  respectively?

Unfortunately, the links provided in post 144 are broken.

**Comment by trivialfis** [comment]
Yes, it's multiplied into gradients. It's also used in some other places like sketching.

**Comment by geogreko** [comment]
Great, thanks for the response!

**Comment by trivialfis** [comment]
Feel free to reopen if there are further questions.

**Comment by geogreko** [comment]
One more question. 
In the case of  sample_weights, do we also multiply the gradients (gi,hi) to the sample_weights when calculating the Gain function? Like g_i * case_weight_i, h_i * case_weight_i, for all gradients in the gain function below?
![image](https://github.com/dmlc/xgboost/assets/76202645/2ce9b35f-83af-4c11-bd72-5c0561809b9f)

**Comment by geogreko** [comment]
Dear Jiaming,

Sorry to disturb you. I tried to reopen the issue, but the button seems deactivated.
I made a comment, but as I am not sure if you are notified, here is my follow-up question:


In the case of sample_weights, do we also multiply the gradients (gi,hi) to the sample_weights when calculating the Gain function? Like g_i * case_weight_i,  h_i * case_weight_i, for all gradients in the gain function below?
[image]<https://private-user-images.githubusercontent.com/76202645/332774805-2ce9b35f-83af-4c11-bd72-5c0561809b9f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTY4NzI4MDQsIm5iZiI6MTcxNjg3MjUwNCwicGF0aCI6Ii83NjIwMjY0NS8zMzI3NzQ4MDUtMmNlOWIzNWYtODNhZi00YzExLWJkNzItNWMwNTYxODA5YjlmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTI4VDA1MDE0NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTIzZTE5MDU0OGVmZGQ1NTAzODhiZTBlM2ZiOTMwN2UyYjBlZGM3YmQ3OTEzMGFiNGEwNmY0ODMyZjA4ODBiNGQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.7BegQZ_LieROeQzoIaovn6O95UllNA4RnvfBZPMx_9o>

Your response is highly appreciated,

Best,
George
________________________________
